### ISSUE TITLE
`<if(o.arrayref)>` for non-null but empty array does not eval as false

### ISSUE THREAD

**Issue by parrt** [antlr/stringtemplate4#169]

**Comment by sharwell** [comment]
Ouch. This isn't what we want, but it seems we're too far into the game on this one (high risk of breaking templates being used in real applications).

**Comment by parrt** [comment]
anyone using it for <o:arrayref:{x | ...}> would have wrong behavior so fixing might not be an issue.

**Comment by sharwell** [comment]
@parrt Wrong behavior is relative after a bug ships. In this case I'm suggesting that there are likely instances of users leveraging the originally-undesired behavior (this bug) as part of a template/application combination that produces the output the expect. Correcting the bug could cause these users to start seeing undesired output from their applications.

**Comment by parrt** [comment]
yeah, it's semi-risky but `<o:arrayref:{x | hi <x>}>` gets `hi `. not sure people would have "used" that functionality.

**Comment by sharwell** [comment]
💭 Maybe make this a minor release (4.1) instead of patch (4.0.9).

The tests should be easy to implement - for every case where we use `List<T>` in `TestNullAndEmptyValues`, add a matching test using `T[]` with the same input and expected output.
